lux/media/lux/lux.js

```diff
@@ -3430,58 +3447,427 @@ angular.module('lux.form.process', ['ngFileUpload'])
+                var defer = $q.defer(),
+                    options = scope[target.name];
+
+                // If the query is searching for value
+                if (searchValue === null)
+                    delete config.params[config.id];
+                else
+                    config.params[config.id] = searchValue;
+
+                if (!extendCurrentOptions) {
+                    // Add initial options
+                    options = [];
+                    scope[target.name] = options;
+                    config.initialValue.id = '';
+                    config.initialValue.repr = 'Loading...';
+                    options.push(config.initialValue);
+                }
+
+                api.get(null, config.params).then(function(data) {
+                    // Get amount of total items
+                    config.optionsTotal = data.data.total;
+
+                    if (searchValue !== null && data.data.result.length === 0) {
+                        options[0].repr = 'No matches found';
+                    } else {
+                        options[0].repr = 'Please select...';
+                    }
 
-    .directive('remoteOptions', ['$lux', function ($lux) {
+                    remoteService.getOptions(data.data.result, options, attrs, config, extendCurrentOptions);
 
+                    // If initially value comes from not the first pagination page,
+                    // then we need to get element from API because of the field repr.
+                    require(['lodash'], function(_) {
+                        var selectedValue = scope[scope.formModelName][attrs.name];
+                        if (!attrs.multiple && searchValue === null) {
+                            var isElementFromFirstPage = _.findIndex(options, function(item) {
+                                return item.id === selectedValue;
+                            });
+
+                            if (isElementFromFirstPage === -1) {
+                                remoteService.excludeValue = selectedValue;
+
+                                config.params[config.id] = selectedValue;
+                                api.get(null, config.params).then(function(data) {
+                                    if (data.data.result.length > 0) {
+                                        var option = remoteService.parseOption(data.data.result[0], attrs, config);
+                                        // Add an option to the third place from the end
+                                        options.splice(options.length-3, 0, option);
+                                        // Update selected value in list
+                                        scope.$select.selected = option;
+                                    }
+                                });
+                            }
+                        }
+                    });
+
+                    defer.resolve(data);
```

Comment:
Should this be resolved here with the require block above? If yes, could we add a comment?